robotoff/workers/tasks/product_updated.py

```diff
@@ -15,21 +16,35 @@
 
 
 @with_db
-def update_insights(barcode: str, server_domain: str):
-    # Sleep 10s to let the OFF update request that triggered the webhook call
-    # to finish
-    logger.info(f"Running `update_insights` for product {barcode} ({server_domain})")
+def update_insights_job(barcode: str, server_domain: str):
+    """This job is triggered by the webhook API, when product information has
+    been updated.
 
-    product_dict = get_product(barcode)
+    When a product is updated, Robotoff will:
 
-    if product_dict is None:
-        logger.warning("Updated product does not exist: %s", barcode)
-        return
+    1. Generate new predictions related to the product's category and name.
+    2. Regenerate all insights from the product associated predictions.
+    """
+    logger.info(f"Running `update_insights` for product {barcode} ({server_domain})")
 
-    updated_product_predict_insights(barcode, product_dict, server_domain)
-    logger.info("Refreshing insights...")
-    imported = refresh_insights(barcode, server_domain)
-    logger.info(f"{imported} insights created after refresh")
+    try:
+        with Lock(
+            name=f"robotoff:product_update_job:{barcode}", expire=300, timeout=10
+        ):
```

Comment:
I think it's worth commenting 
```suggestion
        # We handle concurrency thanks to the lock
        # as the task will fetch product from MongoDB at the time it runs,
        # it's not worth reprocessing with another task arriving concurrently.
        # The expire is there only in case the lock is not released (process killed)
        with Lock(
            name=f"robotoff:product_update_job:{barcode}", expire=300, timeout=10
        ):
```